test/MiddlewareFactoryTest.php

```diff
@@ -24,33 +26,39 @@
 
 class MiddlewareFactoryTest extends TestCase
 {
-    public function setUp()
+    /** @var MiddlewareContainer|ObjectProphecy */
+    private $container;
+
+    /** @var MiddlewareFactory */
+    private $factory;
+
+    protected function setUp()
     {
         $this->container = $this->prophesize(MiddlewareContainer::class);
         $this->factory = new MiddlewareFactory($this->container->reveal());
     }
 
-    public function assertLazyLoadingMiddleware(string $expectedMiddlewareName, MiddlewareInterface $middleware)
+    private function assertLazyLoadingMiddleware(string $expectedMiddlewareName, MiddlewareInterface $middleware)
```

Comment:
Not sure about this. Although these assert functions are only used in this class, I would keep them in line with the phpunit style and make them at least static as well.

Reply:
In PHPUnit style these should be `static`, and it can't be because this requires access to non static property of the class... So definitely the function shouldn't be used outside the class. When we change it to be static then we can make it public again. It's not important, as these are tests only...

Reply:
Again, I'm on the fence here. Test classes are different than package classes. Nobody is going to re-use them, but if they do (and I've extended test cases a number of times), having `private` as the default is a real pain. Testing needs to be as easy as possible. Public visibility is perfectly fine in test classes.